nml/expression/binop.py

```diff
@@ -90,7 +90,7 @@ def reduce(self, id_dicts = [], unknown_id_fatal = True):
         # - If the operator allows it and the second expression is more complex than
         #   the first one swap them.
         op = self.op
-        if op in commutative_operators or self.op in (nmlop.CMP_LT, nmlop.CMP_GT):
+        if op.commutative or op in (nmlop.CMP_LT, nmlop.CMP_GT):
```

Comment:
Would it make sense to define a set (or another property like communative) called 'comparison' or similar?

Reply:
Just a "comparison" flag isn't really helpful, because you'd still need a table of which one to replace it with.

I looked at adding a `swapped_operand_op` property to cover this, but it's awkward because Python doesn't have forward references...

There are several things in this file that could be done more nicely, but I don't think it's particularly in scope for this patch.